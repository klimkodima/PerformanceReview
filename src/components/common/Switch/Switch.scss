.switch-wrapper-external {
  display: flex;
  align-items: center;

  .disabled {
    opacity: 0.7;
    color: #9f9f9f;
  }

  .disabled-switch {
    span {
      cursor: default;
    }
    opacity: 0.7;
    input:checked + .switch {
      background-color: #9f9f9f;
    }
    input + .switch {
      background-color: #9f9f9f;
    }
  }

  p {
    font-weight: bold;
  }
  label {
    position: relative;
    width: 52px;
    height: 26px;
    margin: 0 15px;
  }
  input {
    opacity: 0;
    width: 0;
    height: 0;
  }
  .switch {
    position: absolute;
    cursor: pointer;
    top: 0;
    left: 0;
    right: 0;
    bottom: 0;
    background-color: #ccc;
    transition: 0.4s;
    border-radius: 34px;
  }
  .switch:before {
    position: absolute;
    content: '';
    height: 18px;
    width: 18px;
    left: 4px;
    bottom: 4px;
    background-color: white;
    transition: 0.4s;
    border-radius: 50%;
  }
  input:checked + .switch {
    background-color: #f54470;
  }
  input + .switch {
    background-color: #1e88e5;
  }
  input:checked + .switch:before {
    transform: translateX(26px);
  }
}
