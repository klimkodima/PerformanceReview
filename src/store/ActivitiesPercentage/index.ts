export {
  default as activitiesPercentageReducer,
  fetchAuditActivities,
  setAuditActivities,
  setTeamActivities,
  fetchTeamActivities
} from './reducer';
export { activitiesPercentageSaga } from './sagas';
export { selectAuditActivities, selectTeamActivities } from './selectors';

export type { AuditActivitiesType, ActivityLabelType } from './types';
