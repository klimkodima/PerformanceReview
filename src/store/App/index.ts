export { selectAvailableWidgets, selectSettingsPermission } from './selectors';
export {
  default as appReducer,
  setAvailableWidgets,
  fetchAvailableWidgets
} from './reducer';
export { appSaga } from './sagas';
export type { AvailableWidgetsType, SettingsPermissionType } from './types';
