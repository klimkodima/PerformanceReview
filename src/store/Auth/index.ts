export {
  default as authReducer,
  setAccessToken,
  loginUser,
  setIsAuth,
  setAuthError
} from './reducer';
export type { LoginDataType, CurrentUserType } from './types';
export { AuthSaga } from './sagas';
export {
  selectAuthError,
  selectAuthIsAuth,
  selectCurrentUser
} from './selectors';
