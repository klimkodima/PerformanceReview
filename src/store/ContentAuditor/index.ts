export {
  default as contentAuditorReducer,
  fetchContentAuditor,
  setContentAuditor
} from './reducer';
export { contentAuditorSaga } from './sagas';
export { selectContentAuditor } from './selectors';

export type { ContentAuditorType } from './types';
