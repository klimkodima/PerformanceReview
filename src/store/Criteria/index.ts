export {
  default as criteriaReducer,
  fetchCriteria,
  setCriteria
} from './reducer';
export { criteriaSaga } from './sagas';
export { selectCriteria } from './selectors';
0;
