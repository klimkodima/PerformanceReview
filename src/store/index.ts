import createSagaMiddleware from 'redux-saga';
import { configureStore } from '@reduxjs/toolkit';
import {
  persistStore,
  persistReducer,
  FLUSH,
  REHYDRATE,
  PAUSE,
  PERSIST,
  PURGE,
  REGISTER
} from 'redux-persist';
import storage from 'redux-persist/lib/storage';

import rootReducer from './rootReducer';
import rootSaga from './rootSaga';

const sagaMiddleware = createSagaMiddleware();

const persistConfig = {
  key: 'root',
  storage,
  blacklist: ['dateFilter', 'tasksCoefficient']
};

const persistedReducer = persistReducer(persistConfig, rootReducer);

const store = configureStore({
  reducer: persistedReducer,
  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware({
      serializableCheck: {
        ignoredActions: [FLUSH, REHYDRATE, PAUSE, PERSIST, PURGE, REGISTER]
      }
    }).concat(sagaMiddleware)
});

sagaMiddleware.run(rootSaga);

export type RootStateType = ReturnType<typeof rootReducer>;
export type StoreType = typeof store;

export const persistor = persistStore(store);
export default store;
