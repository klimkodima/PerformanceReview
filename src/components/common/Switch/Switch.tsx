import { ReactElement, ChangeEvent } from 'react';

import './Switch.scss';

type SwitchPropsType = {
  enabledText: string;
  disabledText: string;
  checked: boolean;
  onChange: (e: ChangeEvent<HTMLInputElement>) => void;
  isDisabled: boolean;
};

const Switch = ({
  checked,
  enabledText,
  disabledText,
  onChange,
  isDisabled
}: SwitchPropsType): ReactElement => (
  <div className='switch-wrapper-external'>
    <p className={!checked ? 'disabled' : ''}>{enabledText}</p>
    <label
      className={isDisabled ? 'disabled-switch' : ''}
      data-testid='switch-label-test'
    >
      <input
        type='checkbox'
        checked={!checked}
        onChange={onChange}
        disabled={isDisabled}
        name='switch'
      />
      <span className='switch' />
    </label>
    <p className={checked ? 'disabled' : ''}>{disabledText}</p>
  </div>
);

export default Switch;
