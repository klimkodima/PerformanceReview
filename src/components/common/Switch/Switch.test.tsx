import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';

import store from 'src/store';
import Switch from './Switch';

describe('Switch test', () => {
  test('Switch renders correctly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Switch
            isDisabled={true}
            checked={true}
            onChange={() => true}
            disabledText='Inactive'
            enabledText='Active'
          />
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('Switch disabled text should have class disabled', () => {
    render(
      <Switch
        isDisabled={true}
        checked={true}
        onChange={() => true}
        disabledText='Inactive'
        enabledText='Active'
      />
    );
    expect(screen.getByText('Inactive')).toHaveClass('disabled');
  });

  test('Switch enabled text should have class disabled', () => {
    render(
      <Switch
        isDisabled={true}
        checked={false}
        onChange={() => true}
        disabledText='Inactive'
        enabledText='Active'
      />
    );
    expect(screen.getByText('Active')).toHaveClass('disabled');
  });

  test('Switch should be disabled', () => {
    render(
      <Switch
        isDisabled={true}
        checked={false}
        onChange={() => true}
        disabledText='Inactive'
        enabledText='Active'
      />
    );
    expect(screen.getByRole('checkbox')).toHaveAttribute('disabled');
    expect(screen.getByTestId('switch-label-test')).toHaveClass(
      'disabled-switch'
    );
  });
});
