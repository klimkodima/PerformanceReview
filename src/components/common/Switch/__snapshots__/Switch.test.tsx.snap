// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Switch test Switch renders correctly 1`] = `
<div
  className="switch-wrapper-external"
>
  <p
    className=""
  >
    Active
  </p>
  <label
    className="disabled-switch"
    data-testid="switch-label-test"
  >
    <input
      checked={false}
      disabled={true}
      name="switch"
      onChange={[Function]}
      type="checkbox"
    />
    <span
      className="switch"
    />
  </label>
  <p
    className="disabled"
  >
    Inactive
  </p>
</div>
`;
